libraries/bdp-access-proxy/bdp-access-api-capture/src/main/java/com/bosch/osmi/bdp/access/mock/generator/MockDataGenerator.java

```diff
@@ -0,0 +1,149 @@
+ * 02110-1301, USA.
+ */
+
+package com.bosch.osmi.bdp.access.mock.generator;
+
+import com.bosch.osmi.bdp.access.api.model.BdpEntity;
+import com.bosch.osmi.bdp.access.api.model.User;
+import com.bosch.osmi.bdp.access.impl.BdpApiAccessImpl;
+import com.bosch.osmi.bdp.access.impl.util.Util;
+import com.google.gson.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.Properties;
+
+/**
+ * The MockDataGenerator iterates over all public methods of all api entities and stores the return values in a json
+ * file. It uses the Java Reflection API to inspect the bdp-access-api and google.gson to serialize the result.
+ *
+ * @author muj1be
+ * @since 11/19/15.
+ */
+public class MockDataGenerator {
+
+    private static final Logger LOGGER = LogManager.getLogger(MockDataGenerator.class);
+
+    private String userName;
+    private String password;
+    private String serverUrl;
+
+    public static void main(String[] args) throws IOException, InvocationTargetException, IllegalAccessException {
+        MockDataGenerator generator = new MockDataGenerator();
+        generator.generateMockData();
+    }
+
+    public MockDataGenerator(){
+        try {
+            readProperties();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void readProperties() throws IOException {
+        Properties props = Util.readBdpAccessImplProperties();
+        userName = props.getProperty("user");
+        password = props.getProperty("password");
+        serverUrl= props.getProperty("bdp.url");
+    }
+
+    public void generateMockData() throws IOException, InvocationTargetException, IllegalAccessException {
+        LOGGER.info("Start mock data generation ...");
+        LOGGER.info("Try to initialize API access for user "+ userName + " at server " + serverUrl +".");
+        LOGGER.info("Initialize API access.");
+        BdpApiAccessImpl access = new BdpApiAccessImpl(serverUrl, userName, password);
+
+        LOGGER.info("Retrieve user " + userName);
+        User user = access.retrieveUser();
+        JsonObject resultStore = new JsonObject();
+        LOGGER.info("Start bdp-access-api evaluation");
+        evaluate(user, resultStore);
+
+        writeToJsonFile(resultStore);
+    }
+
+    private void writeToJsonFile(JsonObject resultStore) throws IOException {
+        String path = Util.getHomeDir() + File.separator + ".bdp-access-files" + File.separator + "mockdata.json";
+        File file = new File(path);
+        boolean dirExist = file.getParentFile().mkdirs();
+        if(dirExist){
+            LOGGER.debug("Unable to create directory " + file.getAbsolutePath() + ". Maybe it already exists.");
```

Comment:
minor issue: this will print "Unable to create directory" with the name of the json file. Consider rewording to "Unable to create parent directory for" or correcting to `file.getParentFile().getAbsolutePath()`